src/toil_scripts/adam_pipeline/adam_preprocessing.py

```diff
@@ -81,36 +122,40 @@ def call_adam(masterIP, inputs, arguments):
 def remove_file(masterIP, filename, sparkOnToil):
     """
     Remove the given file from hdfs with master at the given IP address
+
+    :type masterIP: MasterAddress
     """
+    masterIP = masterIP.actual
     if sparkOnToil:
-        containerID = check_output(["ssh", "-o", "StrictHostKeyChecking=no", masterIP, "docker", "ps", \
+        containerID = check_output(["ssh", "-o", "StrictHostKeyChecking=no", masterIP, "docker", "ps",
```

Comment:
Small nit: if I'm splitting a call/array across multiple lines, I'll typically break it out into one line per element/logical group of elements. E.g., this would become something like:

```
containerID = check_output(["ssh",
"-o", "StrictHostKeyChecking=no",
masterIP,
"docker", "ps", "|",
"grep", "apache-hadoop-master", "|",
"awk", "'{print $1}'"])
```

Same for the line below.